File: решения/java/src/Lab14/Comparator7186.java
package Lab14;

import java.util.Comparator;

public class Comparator7186 implements Comparator<Integer> {
  private int x;
  public Comparator7186(int x){
    this.x = x;
  }
  @Override
  public int compare(Integer a, Integer b) {
    int d1 = Math.abs(a - x);
    int d2 = Math.abs(b - x);
    if (d1 > d2)
    {
      return 1;
    }
    else if (d1 == d2)
    {
      return 0;
    }
    else {
      return -1;
    }
  }
}

File: решения/Stage2 cs/CodeProject/Comparator7186.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CodeProject
{
    public class Comparator7186 : Comparer<int>
    {
        private int x;
        public Comparator7186(int x)
        {
            this.x = x;
        }

        public override int Compare(int a, int b)
        {
            int d1 = Math.Abs(a - x);
            int d2 = Math.Abs(b - x);
            if (d1 > d2)
            {
                return 1;
            }
            else if (d1 == d2)
            {
                return 0;
            }
            else
            {
                return -1;
            }
        }
    }
}
